ddl/ddl_worker.go

```diff
@@ -113,20 +107,37 @@ func (d *ddl) updateDDLJob(t *meta.Meta, job *model.Job, updateTS uint64) error
 
 // finishDDLJob deletes the finished DDL job in the ddl queue and puts it to history queue.
 // If the DDL job need to handle in background, it will prepare a background job.
-func (d *ddl) finishDDLJob(t *meta.Meta, job *model.Job) error {
-	log.Infof("[ddl] finish DDL job %v", job)
-	// Job is finished, notice and run the next job.
-	_, err := t.DeQueueDDLJob()
-	if err != nil {
-		return errors.Trace(err)
-	}
+func (d *ddl) finishDDLJob(t *meta.Meta, job *model.Job) (err error) {
 	switch job.Type {
-	case model.ActionDropSchema, model.ActionDropTable, model.ActionTruncateTable:
-		if err = d.prepareBgJob(t, job); err != nil {
+	case model.ActionDropSchema, model.ActionDropTable, model.ActionTruncateTable, model.ActionDropIndex:
+		if job.Version <= currentVersion {
+			if job.Version < bgJobMigrateVersion {
```

Comment:
Because some users maybe skip some versions, upgrade to more new tidb version. In that case, we need `currentVersion`.